Sources/Linq2DynamoDb.DataContext/TableDefinitionWrapperBase.cs

```diff
@@ -46,6 +48,57 @@ internal TableDefinitionWrapperBase(Table tableDefinition, Type tableEntityType,
         /// </summary>
         internal event Action<string> OnLog;
 
+        /// <summary>
+        ///  The name of the underlying entity's property that has the  [DynamoDBVersion] attribute
+        /// </summary>
+        internal string VersionPropertyName 
+        {
+            get 
+            {
+                if (!_hasResolvedVersionPropertyName) 
+                {
+                    var versionProperty = TableEntityType
+                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                        .Where(property =>
+                            property
+                                .GetCustomAttributes(typeof(DynamoDBVersionAttribute), true)
+                                .SingleOrDefault() != null
+                        ).SingleOrDefault();
+
+                    _versionPropertyName = versionProperty != default(PropertyInfo)
+                        ? versionProperty.Name
+                        : null;
+
+                    _hasResolvedVersionPropertyName = true;
```

Comment:
and yeah if the actual `_versionPropertyName` backing field was `Lazy` you wouldn't need the `_hasResolved...` field